db/auto_roll_logger.cc

```diff
@@ -32,8 +32,17 @@ Status AutoRollLogger::ResetLogger() {
 }
 
 void AutoRollLogger::RollLogFile() {
-  std::string old_fname = OldInfoLogFileName(
-      dbname_, env_->NowMicros(), db_absolute_path_, db_log_dir_);
+  uint64_t now = env_->NowMicros();
+  std::string old_fname;
+  // Try to check target name only 10 times at most
+  for (int i = 0; i < 10; i++) {
```

Comment:
It is called when log is rotating. We assume that two rotations can happen quickly and Nowmicro returns same value. To not overwrite file we increment by one micro and try again. We try that 10 times at most to no end up in long cycle. 

I do not see any overhead. Or do we rename logs that often?